modules/apps/company/company-test/src/testIntegration/java/com/liferay/company/service/test/CompanyLocalServiceTest.java

```diff
@@ -562,6 +562,49 @@ public void testAddAndDeleteCompanyWithUserGroupAndUserGroupRole()
 		Assert.assertNull(_userLocalService.fetchUser(user.getUserId()));
 	}
 
+	@Test
+	public void testCheckCompanyDefaultUserLocaleAndTimeZoneDifferFromCompany()
+		throws Exception {
+
+		Company company = addCompany();
+
+		try {
+			User defaultUser = company.getDefaultUser();
+
+			String expectedLanguageId = defaultUser.getLanguageId();
+			String expectedTimeZoneId = defaultUser.getTimeZoneId();
+
+			String langaugeId = "en_US";
+
+			if (langaugeId.equals(expectedLanguageId)) {
+				langaugeId = "fr_FR";
+			}
+
+			String timeZoneId = "UTC";
+
+			if (timeZoneId.equals(expectedTimeZoneId)) {
+				timeZoneId = "GMT";
+			}
+
+			defaultUser.setLanguageId(langaugeId);
```

Comment:
@yuhai 
Steps to reproduce:
1. Start portal from empty database with "company.default.locale=en_US"
2. Shutdown portal and check database table User_, check the default user's language is "en_US"
3. Set "company.default.locale=es_ES" and restart portal
4. check database table User_, check the default user's language is "es_ES"

This doesn't reflect the steps as you are updating the default user by yourself in the test?